lib/Mojo/Collection.pm

```diff
@@ -93,8 +93,11 @@ sub tap { shift->Mojo::Base::tap(@_) }
 sub to_array { [@{shift()}] }
 
 sub uniq {
+  my ($self, $cb) = @_;
   my %seen;
-  return $_[0]->new(grep { !$seen{$_}++ } @{$_[0]});
+  return $self->new(map { (!$seen{ $cb->($_) }++) ? $_ : () } @$self)
```

Comment:
Just a (bad) habit, I use map *far* more than grep, so I use it by default. Fixed.